pkg/policy/rule.go

```diff
@@ -157,8 +140,22 @@ func mergePortProto(ctx *SearchContext, existingFilter, filterToMerge *L4Filter,
 				newL7Rules = &PerSelectorPolicy{}
 			}
 
-			// Merge isRedirect flag
-			l7Rules.isRedirect = l7Rules.isRedirect || newL7Rules.isRedirect
+			// Merge l7 parser type
+			l7Rules.L7Parser, err = l7Rules.L7Parser.Merge(newL7Rules.L7Parser)
+			if err != nil {
+				ctx.PolicyTrace("   Merge conflict: mismatching parsers %s/%s\n", newL7Rules.L7Parser, l7Rules.L7Parser)
+				return err
+			}
+
+			// Merge listener reference
+			if l7Rules.Listener == "" || newL7Rules.Listener == "" {
+				if newL7Rules.Listener != "" {
+					l7Rules.Listener = newL7Rules.Listener
+				}
+			} else if newL7Rules.Listener != l7Rules.Listener {
+				ctx.PolicyTrace("   Merge conflict: mismatching CiliumEnvoyConfig listeners %v/%v\n", newL7Rules.Listener, l7Rules.Listener)
```

Comment:
nit: might be worth to put a space in between the two listener names. the names themselves can already contain slashes - which makes it quite hard to read. -> `%v / %v`

same for the error below, and maybe also the l7parser merge conflict trace on line 146